stdlib/src/builtin/string.mojo

```diff
@@ -1208,43 +1208,46 @@ struct String(
         res.append(0)
         return String(res^)
 
-    fn strip(self) -> String:
-        """Return a copy of the string with leading and trailing whitespace characters removed.
+    fn strip(self, chars: String = " \n\t\r\f\v") -> String:
+        """Return a copy of the string with leading and trailing characters removed.
 
-        See `isspace` for a list of whitespace characters
+        Args:
+            chars: A set of characters to be removed. Defaults to whitespace.
 
         Returns:
-          A copy of the string with no leading or trailing whitespace characters.
+          A copy of the string with no leading or trailing characters.
         """
 
-        return self.lstrip().rstrip()
+        return self.lstrip(chars).rstrip(chars)
 
-    fn rstrip(self) -> String:
-        """Return a copy of the string with trailing whitespace characters removed.
+    fn rstrip(self, chars: String = " \n\t\r\f\v") -> String:
+        """Return a copy of the string with trailing characters removed.
 
-        See `isspace` for a list of whitespace characters
+        Args:
+            chars: A set of characters to be removed. Defaults to whitespace.
 
         Returns:
-          A copy of the string with no trailing whitespace characters.
+          A copy of the string with no trailing characters.
         """
 
         var r_idx = len(self)
-        while r_idx > 0 and isspace(ord(self[r_idx - 1])):
+        while r_idx > 0 and self[r_idx - 1] in chars:
             r_idx -= 1
 
         return self[:r_idx]
 
-    fn lstrip(self) -> String:
-        """Return a copy of the string with leading whitespace characters removed.
+    fn lstrip(self, chars: String = " \n\t\r\f\v") -> String:
+        """Return a copy of the string with leading characters removed.
 
-        See `isspace` for a list of whitespace characters
+        Args:
+            chars: A set of characters to be removed. Defaults to whitespace.
 
         Returns:
-          A copy of the string with no leading whitespace characters.
+          A copy of the string with no leading characters.
         """
 
         var l_idx = 0
-        while l_idx < len(self) and isspace(ord(self[l_idx])):
+        while l_idx < len(self) and self[l_idx] in chars:
```

Comment:
**thoughts** In the default use case, we might end up with a slower function. Before, it was checking that the byte was in a given interval (something like `return a < x < b`, see the implementation of `isspace()`) and now it's checking the value against one of the 5 elements in the list. With simd, the performance might be similar. 

I don't believe it needs to be changed right now with overloads for more performance, I'm just mentionning for the maintainers that we can improve in two areas:
* Ensure the same code path is taken for isspace and the default use of `rstrip`, `lstrip`, to ensure we don't have different behaviors
* Have a benchmarking tool to allow us to verify that we don't have regressions. Here it's not clear if we'll get a performance regression or not. (it might not be the focus this early so it might not be too important).

Reply:
Yep, that will definitely slow down checks a bit, but not so much. But for now it's the only readable and flexible solution at the same time I see.

Reply:
Indeed I'm not asking for changes, it's just a note for the maintainers